platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapView.java

```diff
@@ -174,6 +175,9 @@ private void initialise(@NonNull final Context context, @NonNull final MapboxMap
     // notify Map object about current connectivity state
     nativeMapView.setReachability(ConnectivityReceiver.instance(context).isConnected(context));
 
+    // bind internal components for map change events
+    mapChangeDispatcher.bind(mapCallback = new MapChangeResultHandler(mapboxMap));
```

Comment:
Been looking into it but don't think so at the moment though this clearly shows we need to rethink our architecture. The issue here is a cyclic class dependency: NativeMapview depends on MapChangeDispatcher, MapboxMap depends on NativeMapView and MapChangeDispatcher depends on MapboxMap. 

Somewhere during initialization we need to bind them together and require us to have null checking. Current setup follows the requirement of being able to listen to map change events as soon as possible. I don't see an a issue with internal components missing out on the initialization change events.